akka-http-core/src/main/java/akka/http/javadsl/model/AttributeKeys.java

```diff
@@ -6,9 +6,13 @@
 
 import akka.http.javadsl.model.ws.WebSocketUpgrade;
 
+import javax.net.ssl.SSLSession;
+
 public final class AttributeKeys {
     public static final AttributeKey<RemoteAddress> remoteAddress =
             (AttributeKey<RemoteAddress>)(Object)akka.http.scaladsl.model.AttributeKeys.remoteAddress();
     public static final AttributeKey<WebSocketUpgrade> webSocketUpgrade =
             (AttributeKey<WebSocketUpgrade>)(Object)akka.http.scaladsl.model.AttributeKeys.webSocketUpgrade();
+    public static final AttributeKey<SSLSession> sslSession =
```

Comment:
This cast is so sneaky that I don't think it will even work ;)

If a value class instance is used in a generic environment, it will be/stay boxed, so retrieving the value like this will likely fail.

Reply:
You're quite right ;). I guess we have to wrap the session for both Java and Scala.